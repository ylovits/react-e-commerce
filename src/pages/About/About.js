import React, { Component } from "react";
import ReactMarkdown from "react-markdown";

const AppMarkdown = require("../../README.md");

class About extends Component {
  constructor() {
    super();
    this.state = { markdown: "" };
  }

  componentDidMount() {
    fetch(AppMarkdown)
      .then(res => res.text())
      .then(text => this.setState({ markdown: text }));
  }
  render() {
    const { markdown } = this.state;
    return (
      <div className="bg-dark text-light about mb-4 p-4">
        <ReactMarkdown source={markdown} />
      </div>
    );
  }
}

export default About;
