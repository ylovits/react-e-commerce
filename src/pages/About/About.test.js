import React from "react";
import { shallow } from "enzyme";
import About from "./About";

it("<About /> rendered", () => {
  expect(shallow(<About />)).toMatchSnapshot();
});
