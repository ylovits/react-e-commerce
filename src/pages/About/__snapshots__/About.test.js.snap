// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> rendered 1`] = `ShallowWrapper {}`;
